src/core/backbone_router/local.cpp

```diff
@@ -238,6 +247,114 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
     return;
 }
 
+otError Local::GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+
+    aConfig = mDomainPrefixConfig;
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength == aPrefix.mLength, error = OT_ERROR_NOT_FOUND);
+
+    VerifyOrExit(Ip6::Address::PrefixMatch(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8, aPrefix.mPrefix.mFields.m8,
+                                           BitVectorBytes(aPrefix.mLength)) >= aPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    if (IsEnabled())
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength);
+        LogDomainPrefix("Remove", error);
+    }
+
+    mDomainPrefixConfig.mPrefix.mLength = 0;
+
+exit:
+    return error;
+}
+
+void Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error; // only used for logging.
+
+    if (IsEnabled() && mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+                                                             mDomainPrefixConfig.mPrefix.mLength);
+
+        LogDomainPrefix("Remove", error);
+    }
+
+    mDomainPrefixConfig = aConfig;
+    LogDomainPrefix("Set", OT_ERROR_NONE);
+
+    if (IsEnabled())
+    {
+        error = Get<NetworkData::Local>().AddOnMeshPrefix(mDomainPrefixConfig);
+        LogDomainPrefix("Add", error);
+    }
+}
+
+void Local::RemoveDomainPrefixFromNetworkData(void)
+{
+    otError error = OT_ERROR_NOT_FOUND; // only used for logging.
+
+    if (mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+
+                                                             mDomainPrefixConfig.mPrefix.mLength);
+        OT_ASSERT(error == OT_ERROR_NONE);
```

Comment:
I think `OT_ASSERT` here is too strict because we can not guarantee that `AddDomainPrefixToNetworkData` always succeed. `AddDomainPrefixToNetworkData` can fail due to `NetworkDataTlv ` overflow, or even conflicting with `ML prefix`. So it is possible that backbone router is enabled and `mDomainPrefixConfig.mPrefix.mLength > 0`, but domain prefix is not in network data?

Reply:
there are already some pre-checks when set domain prefix - already checked whether or not conflicts or overflow, so in general, the prefix should always be added successfully when backbone functionality was enabled if there are not many other server configurations before enable.

Reply:
removed the assert